src/v/datalake/errors.h

```diff
@@ -12,6 +12,13 @@
 #include <stdexcept>
 namespace datalake {
 
+enum class coordinator_errc : int16_t {
```

Comment:
are we going to have a fully separate class of errors for coordinator ? would it be useful to make them translatable to `std::error_code` by adding an error category ?